core/trino-main/src/main/java/io/trino/metadata/AbstractPropertyManager.java

```diff
@@ -69,28 +75,211 @@ protected final void doRemoveProperties(K propertiesKey)
         connectorProperties.remove(propertiesKey);
     }
 
+    protected final Properties doGetOnlySpecifiedProperties(
+            K propertiesKey,
+            String catalogNameForDiagnostics,
+            Iterable<Property> properties,
+            Session session,
+            Metadata metadata,
+            AccessControl accessControl,
+            Map<NodeRef<Parameter>, Expression> parameters)
+    {
+        Map<String, PropertyMetadata<?>> supportedPropertiesMetadata = getSupportedPropertiesMetadata(propertiesKey, catalogNameForDiagnostics);
+
+        SeparatedProperties separatedProperties = separateProperties(propertiesKey, catalogNameForDiagnostics, properties);
+
+        // builder for properties whose values (after being evaluated or looked up) are non-null
```

Comment:
Should it be something like
`builder for properties for which values ... are null`?